models/models.go

```diff
@@ -254,3 +254,28 @@ func MaxBatchInsertSize(bean interface{}) int {
 func Count(bean interface{}) (int64, error) {
 	return x.Count(bean)
 }
+
+// IsTableNotEmpty returns true if table has at least one record
+func IsTableNotEmpty(tableName string) (bool, error) {
+	return x.Table(tableName).Exist()
+}
+
+// DeleteAllRecords will delete all the records of this table
+func DeleteAllRecords(tableName string) error {
```

Comment:
Here too we should also use `beanOrTableName interface{}` if possible.

BTW, I think we should use `TRUNCATE TABLE` because we're not currently using a transaction. Otherwise it will be disruptive because the operation may fail due to table locks, or make others fail.

Reply:
@guillep2k Not all database support that SQL. i.e. https://www.techonthenet.com/sqlite/truncate.php, sqlite didn't support.

Reply:
> 
> 
> @guillep2k Not all database support that SQL. i.e. https://www.techonthenet.com/sqlite/truncate.php, sqlite didn't support.

Oh, OK. We don't need to TRUNCATE a table used by multiple transactions at the same time, after all.